docs/getting-started.md

```diff
@@ -34,221 +37,357 @@ The high level data flow is as follows:
 
-## 4 Build
+## 4 Build and Deploy your Subgraph 
 
 ### 4.1 Build and deploy your mappings to IPFS
 In order to deploy your subgraph to your Graph Node, the subgraph manifest will first need to be built and deployed to IPFS (along with all linked files).
 
-Follow the instructions [here](https://ipfs.io/docs/getting-started/) to start a locally running IPFS daemon.
+To get an IPFS daemon working locally,do the following:
+* Download and install IPFS 
+* run `ipfs init`
+* run `ipfs daemon`
+
+If you get stuck, you can follow the instructions from the IPFS website [here](https://ipfs.io/docs/getting-started/).
+
+Once you've started your IPFS daemon you can run `yarn build-ipfs` in your subgraph directory (it assumes the default tcp port for the IPFS daemon). `yarn build-ipfs` should have been set up in section 3.3 when you went through the [graph-cli documentation](https://github.com/graphprotocol/graph-cli). This will compile your mappings, and deploy the mappings, schema and the subgraph manifest itself to IPFS. The result should be a single content hash, and this is your `subgraph ID`. It will also create a `/dist` folder. There is no need to upload this folder to your git repository, and you should not edit any of the files in there because they are auto generated. 
+
+To confirm on  ipfs, you can pass that content hash into `ipfs cat` to view your subgraph manifest with files paths replaced by IPLD links.
+
+### 4.2 Deploying Your Subgraph
+
+Deploying your subgraph means the subgraph is running, and sourcing data from Ethereum, transforming that data with your mappings, and storing it in the graph node. Note that a running subgraph can safely be stopped and started, and it will pick up where it has left off. 
+
+#### 4.2.1 Create your Postgres db 
+
+Make sure you have postgres installed. Then run the following commands:
+
+`initdb -D .postgres`
+`createdb <YOUR_POSTGRES_DB_NAME>`
+
+Name your db something relevant to your project, so you always know how to access it. 
+
+#### 4.2.3 Setting up your Ethereum Endpoint
+
+When you deploy your subgraph, you need to tell the subgraph where to start grabbing Ethereum data from. There are three typical ways you can grab data from Ethereum which the graph supports, which are Infura, local nodes, and ganache. 
+
+Note, the command you must run requires that you have `rust` and `cargo` installed.
+
+##### 4.2.3.1 Infura
+
+[Infura](https://infura.io/) is supported for both websocket and http connections. This is the simplest way to source mainnet or testnet data, as you don't have to set up your own geth or parity node. However, it does sync slower than being connected to your own node. The following commands are passed to the `graph` command to indicate you want to use infura:
+
+*Web Socket*
+```
+cargo run -p graph-node --release -- \
+  --postgres-url postgresql://<USERNAME><:PASSWORD>@localhost:5432/<YOUR_POSTGRES_DB_NAME> \
```

Comment:
PASSWORD is optional, and I am not sure how to indicate that. We had it as [:PASSWORD], but we didn't use <> :) 
Also, I am still not sure if we want to use <YOUR_POSTGRES_DB_NAME> or just <POSTGRES_DB_NAME> and <YOUR_SUBGRAPH_NAME> or <SUBGRAPH_NAME>. Maybe we should have others weigh in on what's our convention, and then do it consistently. Just a thought

Reply:
probably better without YOUR. we should get a convention though

passowrd is optional in some cases i believe. one user opened an issue, apparently he needed the password for linux. or else if you setup your postgres to be password protected it might want it. it seems like it can go either way